frame/ranked-collective/src/lib.rs

```diff
@@ -0,0 +1,539 @@
+		type AdminOrigin: EnsureOrigin<Self::Origin>;
+
+		/// The polling system used for our voting.
+		type Polls: Polling<
+			TallyOf<Self, I>,
+			Votes = Votes,
+			Class = Rank,
+			Moment = Self::BlockNumber,
+		>;
+
+		/// Convert a rank_delta into a number of votes the rank gets.
+		///
+		/// Rank_delta is defined as the number of ranks above the minimum required to take part
+		/// in the poll.
+		type VoteWeight: Convert<Rank, Votes>;
+	}
+
+	/// The number of members in the collective who have at least the rank according to the index
+	/// of the vec.
+	#[pallet::storage]
+	pub type MemberCount<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Twox64Concat, Rank, MemberIndex, ValueQuery>;
+
+	/// The current members of the collective.
+	#[pallet::storage]
+	pub type Members<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Twox64Concat, T::AccountId, MemberRecord>;
+
+	/// The index of each ranks's member into the group of members who have at least that rank.
+	#[pallet::storage]
+	pub type IdToIndex<T: Config<I>, I: 'static = ()> =
+		StorageDoubleMap<_, Twox64Concat, Rank, Twox64Concat, T::AccountId, MemberIndex>;
+
+	/// The members in the collective by index. All indices in the range `0..MemberCount` will
+	/// return `Some`, however a member's index is not guaranteed to remain unchanged over time.
+	#[pallet::storage]
+	pub type IndexToId<T: Config<I>, I: 'static = ()> =
+		StorageDoubleMap<_, Twox64Concat, Rank, Twox64Concat, MemberIndex, T::AccountId>;
+
+	/// Votes on a given proposal, if it is ongoing.
+	#[pallet::storage]
+	pub type Voting<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		PollIndexOf<T, I>,
+		Twox64Concat,
+		T::AccountId,
+		VoteRecord,
+	>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config<I>, I: 'static = ()> {
+		/// A member `who` has been added.
+		MemberAdded { who: T::AccountId },
+		/// The member `who`'s rank has been changed to the given `rank`.
+		RankChanged { who: T::AccountId, rank: Rank },
+		/// The member `who` of given `rank` has been removed from the collective.
+		MemberRemoved { who: T::AccountId, rank: Rank },
+		/// The member `who` has voted for the `poll` with the given `vote` leading to an updated
+		/// `tally`.
+		Voted { who: T::AccountId, poll: PollIndexOf<T, I>, vote: VoteRecord, tally: TallyOf<T, I> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T, I = ()> {
+		/// Account is already a member.
+		AlreadyMember,
+		/// Account is not a member.
+		NotMember,
+		/// The given poll index is unknown or has closed.
+		NotPolling,
+		/// The given poll is still ongoing.
+		Ongoing,
+		/// There are no further records to be removed.
+		NoneRemaining,
+		/// Unexpected error in state.
+		Corruption,
+		/// The member's rank is too low to vote.
+		RankTooLow,
+		/// The information provided is incorrect.
+		InvalidWitness,
+	}
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
```

Comment:
I don't seem to see an extrinsic to create a poll, only the ability to `vote`.

Reply:
Yes, polls can be created by anyone in the relevant Referenda pallet.